OpenRA.Mods.Common/Activities/Air/TakeOff.cs

```diff
@@ -12,48 +12,89 @@
 using System;
 using OpenRA.Activities;
 using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Activities
 {
 	public class TakeOff : Activity
 	{
 		readonly Aircraft aircraft;
 		readonly IMove move;
-		Func<Actor, Activity, CPos, bool> moveToRallyPoint;
+		readonly Target target;
+		bool moveToRallyPoint;
 
-		public TakeOff(Actor self)
+		public TakeOff(Actor self, Target target)
 		{
 			aircraft = self.Trait<Aircraft>();
 			move = self.Trait<IMove>();
-			moveToRallyPoint = (actor, activity, pos) => NextActivity == null;
+			this.target = target;
+		}
+
+		public TakeOff(Actor self)
+			: this(self, Target.FromCell(self.World, self.Location))
+		{
+			var host = aircraft.GetActorBelow();
```

Comment:
Is this save to do? We had issues before with not doing such in `OnFirstRun`.

Reply:
For the record:
```
02:04:08 (abcdefg30) so you agree that we can merge the PR once the OnFirstRun issue is solved?
02:04:35 (abcdefg30) I am kind of sure you can provoke breakage with that stuff being calculated in the ctor, but didn't construct a case (yet)
02:05:47 (pchote) yeah
02:06:00 (pchote) as much as i want to see the end of that PR, doing that in the ctor is bad
02:06:17 (pchote) but then lets merge ASAP once its fixed
```

Reply:
I just tried moving it to OnFirstRun locally, now the activity is broken in such a ridiculous way I don't even know how to debug it.
Pushed the changes so someone can take a look, but if nobody manages to debug that quickly, please let's settle for leaving it in the ctor and add a TODO for now.

Reply:
> I just tried moving it to OnFirstRun locally, now the activity is broken in such a ridiculous way I don't even know how to debug it.

I debugged this by `Console.WriteLine`ing the `rallyPointDestination` set in `OnFirstRun`.

The first run sets the correct value, but it failed and set `self.Location` the second time.
Wait, what? `OnFirstRun` should not run twice... Further investigation showed that `aircraft.UnReserve` was unconditionally queuing a fresh `TakeOff` that then broke things.

Adding a flag to `aircraft.UnReserve` to disable this fixed the problem.